authui/src/phoneInput.ts

```diff
@@ -13,74 +18,132 @@ interface PhoneInputCountry {
+
+    const preferredCountries: CountryCode[] =
+      JSON.parse(
+        document
+          .querySelector("meta[name=x-phone-input-preferred-countries]")
+          ?.getAttribute("content") ?? "null"
+      ) ?? [];
+    let initialCountry: CountryCode | null = null;
+
+    const defaultValue = this.inputTarget.value;
+    if (defaultValue) {
+      let phoneInputValue = defaultValue;
+      const defaultPhone = new AsYouType();
+      defaultPhone.input(defaultValue);
+      initialCountry = defaultPhone.getCountry() ?? null;
+      if (initialCountry != null) {
+        const callingCode = "+" + getCountryCallingCode(initialCountry);
+        phoneInputValue = phoneInputValue.replace(callingCode, "");
+      }
+      this.phoneInputTarget.value = phoneInputValue;
+    }
+
+    initialCountry ??=
+      (document
+        .querySelector("meta[name=x-geoip-country-code]")
+        ?.getAttribute("content") as CountryCode) ?? null;
+
     const lang = document.documentElement.lang || "en";
-    const countryCodes = getCountries();
+    const countryCodesMap: Map<CountryCode, null> = new Map();
+    for (const preferredCountry of preferredCountries) {
+      if (countryCodesMap.has(preferredCountry)) {
+        continue;
+      }
+      countryCodesMap.set(preferredCountry, null);
+    }
+    for (const onlyCountry of onlyCountries) {
+      if (countryCodesMap.has(onlyCountry)) {
+        continue;
+      }
+      countryCodesMap.set(onlyCountry, null);
+    }
+    const countryCodes: CountryCode[] = Array.from(countryCodesMap.keys());
+
     const localizedTerritories = territoriesMap[lang];
     const territories =
       localizedTerritories?.main[lang as keyof typeof localizedTerritories.main]
         ?.localeDisplayNames.territories ||
       defaultTerritories.main.en.localeDisplayNames.territories;
 
-    this._countries = countryCodes
-      .map((countryCode) => {
-        const countryLocalizedName = territories[countryCode];
-        const countryName =
-          defaultTerritories.main.en.localeDisplayNames.territories[
-            countryCode
-          ];
-        const countryFlag = getEmojiFlag(countryCode);
-        const countryCallingCode = getCountryCallingCode(countryCode);
-        return {
-          flagEmoji: countryFlag,
-          localizedName: countryLocalizedName,
-          name: countryName,
-          iso2: countryCode,
-          phone: countryCallingCode,
-        };
-      })
-      .sort((a, b) => a.name.localeCompare(b.name));
+    this._countries = countryCodes.map((countryCode) => {
+      const countryLocalizedName = territories[countryCode];
+      const countryName =
+        defaultTerritories.main.en.localeDisplayNames.territories[countryCode];
+      const countryFlag = getEmojiFlag(countryCode);
+      const countryCallingCode = getCountryCallingCode(countryCode);
+      return {
+        flagEmoji: `<span class="country-flag-icon phone-input__country-flag">${countryFlag}</span>`,
```

Comment:
Need to add to tailwind safelist?

Reply:
I included `"./src/**/*.{css,scss,ts}"` in tailwind content config, so I think it should catch call classes in typescript files. (btw this line has no tailwind classes)